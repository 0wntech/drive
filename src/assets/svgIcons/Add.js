import * as React from 'react';

function SvgAdd(props) {
    return (
        <svg width={29} height={28} fill="none" {...props}>
            <ellipse
                cx={14.74}
                cy={13.901}
                rx={13.948}
                ry={13.901}
                fill="#36A510"
            />
            <path
                d="M22.901 14c0 .6-.402 1-1.005 1h-6.03v6c0 .6-.402 1-1.005 1-.603 0-1.005-.4-1.005-1v-6h-6.03c-.602 0-1.004-.4-1.004-1s.402-1 1.005-1h6.03V7c0-.6.401-1 1.004-1s1.005.4 1.005 1v6h6.03c.603 0 1.005.4 1.005 1z"
                fill="#fff"
            />
            <mask
                id="add_svg__a"
                maskUnits="userSpaceOnUse"
                x={6}
                y={6}
                width={17}
                height={16}
            >
                <path
                    d="M22.901 14c0 .6-.402 1-1.005 1h-6.03v6c0 .6-.402 1-1.005 1-.603 0-1.005-.4-1.005-1v-6h-6.03c-.602 0-1.004-.4-1.004-1s.402-1 1.005-1h6.03V7c0-.6.401-1 1.004-1s1.005.4 1.005 1v6h6.03c.603 0 1.005.4 1.005 1z"
                    fill="#fff"
                />
            </mask>
            <g mask="url(#add_svg__a)">
                <path
                    fillRule="evenodd"
                    clipRule="evenodd"
                    d="M2.802 2h24.119v24H2.802V2z"
                    fill="#fff"
                />
            </g>
        </svg>
    );
}

export default SvgAdd;
