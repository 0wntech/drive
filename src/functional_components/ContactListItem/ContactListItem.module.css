.container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    border: 1px solid #c4c4c4;
    box-sizing: border-box;
    border-radius: 20px;
    height: 185px;
    padding: 0 20px;
    font-size: 18px;
    padding: 35px 10px 16px 10px;
    cursor: pointer;
    position: relative;
}

.imageContainer {
    width: 80px;
    height: 80px;
}

.iconContainer {
    position: absolute;
    top: 14px;
    right: 14px;
    cursor: pointer;
}

.iconContainer:hover ellipse {
    fill: #ff697b;
}

.image {
    width: 80px;
    height: 80px;
    border-radius: 50%;
    background-size: cover;
    background-position: center;
}

.nameContainer {
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
    text-align: center;
    margin-top: 30px;
    width: 80%;
}
