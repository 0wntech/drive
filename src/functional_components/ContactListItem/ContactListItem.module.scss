@import '../../../styles/constants.scss';

.container {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    width: 100%;
    border: 10px solid $dark_grey;
    box-sizing: border-box;
    border-radius: 20px;
    height: 180px;
    padding: 1em;
    position: relative;
}

.imageContainer {
    width: 80px;
    height: 80px;
    cursor: pointer;
}

.iconContainer {
    position: absolute;
    top: 14px;
    right: 14px;
    cursor: pointer;
}

.iconContainer:hover ellipse {
    fill: $danger_red;
}

.iconContainer ellipse {
    fill: $light_grey;
}

.iconContainer.added:hover ellipse {
    fill: #36a510;
}

.iconContainer.added ellipse {
    fill: #36c510;
}

.image {
    width: 80px;
    height: 80px;
    border-radius: 50%;
    background-size: cover;
    background-position: center;
}

.nameContainer {
    white-space: pre-wrap;
    word-break: keep-all;
    text-align: center;
    margin-top: .5em;
    max-width: 80%;
    cursor: pointer;
}
