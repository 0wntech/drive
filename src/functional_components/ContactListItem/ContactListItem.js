import React, { useState } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './ContactListItem.module.scss';
import Add from '../../assets/svgIcons/Add';
import Delete from '../../assets/svgIcons/Delete';
import { getInitialsFromUser } from '../../utils/helper';
import DefaultIcon from '../DefaultIcon';
import { getUsernameFromWebId } from '../../utils/url';

const ContactListItem = ({
    contact,
    webId,
    className,
    addContact,
    removeContact,
    removable,
    onClick,
    isContact,
}) => {
    const [wasAdded, setWasAdded] = useState(isContact);
    const initials = getInitialsFromUser(contact);
    return (
        <div
            className={classNames(styles.container, className)}
            onClick={() => onClick(contact)}
        >
            <div
                className={styles.imageContainer}
                onClick={() => onClick(contact)}
                data-test-id="contact-picture"
            >
                {contact.picture ? (
                    <div
                        className={styles.image}
                        style={{ backgroundImage: `url('${contact.picture}')` }}
                    />
                ) : (
                    <DefaultIcon
                        initials={initials}
                        className={styles.defaultIcon}
                    />
                )}
            </div>
            <div className={styles.nameContainer}>
                {contact.name
                    ? contact.name
                    : getUsernameFromWebId(contact.webId)}
            </div>
            <div
                className={classNames(styles.iconContainer, {
                    [styles.added]: !wasAdded,
                })}
            >
                {removable &&
                    (wasAdded ? (
                        <Delete
                            onClick={(e) => {
                                removeContact(webId, contact);
                                if (wasAdded) {
                                    setWasAdded(false);
                                }
                                e.stopPropagation();
                            }}
                        />
                    ) : (
                        <Add
                            onClick={(e) => {
                                addContact(webId, contact);
                                if (!wasAdded) {
                                    setWasAdded(true);
                                }
                                e.stopPropagation();
                            }}
                        />
                    ))}
            </div>
        </div>
    );
};

ContactListItem.propTypes = {
    contact: PropTypes.object,
    className: PropTypes.string,
};

export default ContactListItem;
