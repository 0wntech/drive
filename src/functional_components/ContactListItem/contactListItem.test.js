import React from 'react';
import TestRenderer from 'react-test-renderer';
import ContactListItem from './ContactListItem';
import { contacts } from '../../../test/fixtures/contacts';
import { shallow } from 'enzyme';
import Add from '../../assets/svgIcons/Add';

describe('ContactListItem', () => {
    test('should render correctly', () => {
        const contact = contacts[0];
        const webId = 'https://bejow.solid.community/profile/card#me';
        const className = 'test';
        const addContact = jest.fn();
        const onClick = jest.fn();

        const tree = TestRenderer.create(
            <ContactListItem
                contact={contact}
                webId={webId}
                className={className}
                addContact={addContact}
                onClick={onClick}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('should call addContact', () => {
        const contact = contacts[0];
        const webId = 'https://bejow.solid.community/profile/card#me';
        const className = 'test';
        const addContact = jest.fn();
        const onClick = jest.fn();

        const wrapper = shallow(
            <ContactListItem
                contact={contact}
                webId={webId}
                className={className}
                addContact={addContact}
                onClick={onClick}
            />
        );
        wrapper.find(Add).simulate('click', { stopPropagation: () => {} });
        expect(addContact.mock.calls.length).toEqual(1);
    });

    test('should not call onClick when clicking on addContact', () => {
        const contact = contacts[0];
        const webId = 'https://bejow.solid.community/profile/card#me';
        const className = 'test';
        const addContact = jest.fn();
        const onClick = jest.fn();

        const wrapper = shallow(
            <ContactListItem
                contact={contact}
                webId={webId}
                className={className}
                addContact={addContact}
                onClick={onClick}
            />
        );
        wrapper.find(Add).simulate('click', { stopPropagation: () => {} });
        expect(onClick.mock.calls.length).toEqual(0);
    });

    test('should call onClick', () => {
        const contact = contacts[0];
        const webId = 'https://bejow.solid.community/profile/card#me';
        const className = 'test';
        const addContact = jest.fn();
        const onClick = jest.fn();

        const wrapper = shallow(
            <ContactListItem
                contact={contact}
                webId={webId}
                className={className}
                addContact={addContact}
                onClick={onClick}
            />
        );
        wrapper.find('[data-test-id="contact-picture"]').simulate('click');
        expect(onClick.mock.calls.length).toEqual(1);
    });
});
