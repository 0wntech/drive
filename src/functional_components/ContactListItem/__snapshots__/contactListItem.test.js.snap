// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactListItem should render correctly 1`] = `
<div
  className="container test"
  data-test-id="contact-ludwigschubert"
  onClick={[Function]}
>
  <div
    className="imageContainer"
    data-test-id="contact-picture"
    onClick={[Function]}
  >
    <div
      className="image"
    >
      <img
        onError={[Function]}
        src="https://ludwigschubert.solid.community/profile/aa.jpeg"
      />
    </div>
  </div>
  <div
    className="nameContainer"
  >
    Ludwig Faust Schubert
  </div>
  <div
    className="iconContainer added"
  />
</div>
`;
